File: App07NavigationDrawer/app/src/main/java/com/udemyandroid/app07navigationdrawer/MainActivity.java
package com.udemyandroid.app07navigationdrawer;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.view.View;
import android.view.Menu;

import com.google.android.material.snackbar.Snackbar;
import com.google.android.material.navigation.NavigationView;

import androidx.navigation.NavController;
import androidx.navigation.Navigation;
import androidx.navigation.ui.AppBarConfiguration;
import androidx.navigation.ui.NavigationUI;
import androidx.drawerlayout.widget.DrawerLayout;
import androidx.appcompat.app.AppCompatActivity;

import com.udemyandroid.app07navigationdrawer.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

    private AppBarConfiguration mAppBarConfiguration;
    private ActivityMainBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        setSupportActionBar(binding.appBarMain.toolbar);
        binding.appBarMain.fab.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                enviarEmail();
            }
        });
        DrawerLayout drawer = binding.drawerLayout;
        NavigationView navigationView = binding.navView;
        // Passing each menu ID as a set of Ids because each
        // menu should be considered as top level destinations.
        mAppBarConfiguration = new AppBarConfiguration.Builder(
                R.id.nav_principal, R.id.nav_servico, R.id.nav_clientes,
                R.id.nav_contato, R.id.nav_sobre
                )
                .setOpenableLayout(drawer)
                .build();
        NavController navController = Navigation.findNavController(this, R.id.nav_host_fragment_content_main);
        NavigationUI.setupActionBarWithNavController(this, navController, mAppBarConfiguration);
        NavigationUI.setupWithNavController(navigationView, navController);
    }

    public void enviarEmail(){
        String celular = "tel:2192345678";
        String imagem = "https://http2.mlstatic.com/D_NQ_NP_831464-MLB31136887903_062019-O.webp";
        //Intent intent = new Intent( Intent.ACTION_DIAL, Uri.parse("tel:2192345678") );
        //Intent intent = new Intent( Intent.ACTION_VIEW, Uri.parse(imagem) );
        Intent intent = new Intent( Intent.ACTION_SEND );

        intent.putExtra( Intent.EXTRA_EMAIL, new String []{"android.studio@android.com"});
        intent.putExtra( Intent.EXTRA_SUBJECT, "Contato pelo App - Claudio");
        intent.putExtra( Intent.EXTRA_TEXT, "Mensagem Automática Gerada Pelo App");

        intent.setType("text/plain");

        startActivity( Intent.createChooser(intent,"Compartilhar") );
    }


    @Override
    public boolean onSupportNavigateUp() {
        NavController navController = Navigation.findNavController(this, R.id.nav_host_fragment_content_main);
        return NavigationUI.navigateUp(navController, mAppBarConfiguration)
                || super.onSupportNavigateUp();
    }
}

File: GastoViagem/app/src/main/java/com/claudio/udemycourse/gastoviagem/MainActivity.kt
package com.claudio.udemycourse.gastoviagem

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Toast
import com.claudio.udemycourse.gastoviagem.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity(), View.OnClickListener {

    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        // View Binding
        binding = ActivityMainBinding.inflate(layoutInflater)
        // Initiate
        setContentView(binding.root)
        binding.buttonCalculate.setOnClickListener(this)
    }

    override fun onClick(view: View) {
        if (view.id == R.id.button_calculate){
            calculate()
        }
    }

    private fun isValid(): Boolean{
        return (   binding.editDistance.text.toString() != ""
                && binding.editPrice.text.toString() != ""
                && binding.editAuthonomy.text.toString() != ""
                && binding.editAuthonomy.text.toString().toFloat() != 0f)
    }

    private fun calculate(){
        if (isValid()) {
            val distance = binding.editDistance.text.toString().toFloat()
            val price = binding.editPrice.text.toString().toFloat()
            val authonomy = binding.editAuthonomy.text.toString().toFloat()

            val totalValue = (distance * price) / authonomy
            binding.textTotalValue.text = "R$${"%.2f".format(totalValue)}"
        } else {
            Toast.makeText(this, R.string.validate_all_fields, Toast.LENGTH_SHORT).show()
        }

    }


}